File: backend/mlb/mlb/shared/upsert_player_id.py
# File: backend/scripts/shared/upsert_player_id.py

import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv()

SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_SERVICE_ROLE_KEY = os.getenv("SUPABASE_SERVICE_ROLE_KEY")
supabase: Client = create_client(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY)

async def upsert_player_id(player_name: str, team: str) -> str:
    """
    Resolve and upsert player_id for a given player_name and team from Supabase.
    """
    try:
        # Step 1: Try player_ids
        result = (
            supabase.table("player_ids")
            .select("player_id")
            .eq("player_name", player_name)
            .eq("team", team)
            .maybe_single()
            .execute()
        )

        if result.data and result.data.get("player_id"):
            return result.data["player_id"]

        # Step 2: Fallback to model_training_props
        fallback = (
            supabase.table("model_training_props")
            .select("player_id")
            .eq("player_name", player_name)
            .eq("team", team)
            .order("game_date", desc=True)
            .limit(1)
            .maybe_single()
            .execute()
        )

        player_id = fallback.data.get("player_id") if fallback.data else None
        if not player_id:
            raise ValueError(f"Unable to find player_id for {player_name} ({team})")

        # Step 3: Upsert into player_ids
        supabase.table("player_ids").upsert(
            {
                "player_name": player_name,
                "team": team,
                "player_id": player_id,
            },
            on_conflict="player_id",
        ).execute()

        return player_id

    except Exception as e:
        raise RuntimeError(f"upsert_player_id failed: {e}")

File: backend/mlb/shared/upsertPlayerID.js
// backend/scripts/shared/upsertPlayerID.js

export async function upsertPlayerID(supabase, { player_name, team }) {
  // Check if player already exists
  const { data: existing, error: fetchError } = await supabase
    .from("player_ids")
    .select("player_id")
    .eq("player_name", player_name)
    .eq("team", team)
    .limit(1);

  if (existing?.length) return existing[0].player_id;

  // Fallback: check model_training_props
  const { data: mtp, error: mtpError } = await supabase
    .from("model_training_props")
    .select("player_id")
    .eq("player_name", player_name)
    .eq("team", team)
    .limit(1);

  if (!mtp?.length) {
    console.warn(
      `❌ upsertPlayerID: Could not resolve player_id for ${player_name} (${team})`
    );
    return null;
  }

  const player_id = mtp[0].player_id;

  // Upsert into player_ids
  const { error: upsertError } = await supabase.from("player_ids").upsert(
    {
      player_name,
      player_id,
      team,
    },
    { onConflict: "player_id" }
  );

  if (upsertError) {
    console.warn(
      `⚠️ upsertPlayerID failed for ${player_name}:`,
      upsertError.message
    );
  } else {
    console.log(`✅ upserted player_id for ${player_name}: ${player_id}`);
  }

  return player_id;
}
